torchao/quantization/prototype/mixed_precision/scripts/mp_quant_eval.py

```diff
@@ -0,0 +1,125 @@
+import torch
+import torch.nn as nn
+
+from naive_intNwo import intN_weight_only_asym, intN_weight_only_sym
+from transformers import AutoModelForCausalLM, AutoTokenizer
+
+from lm_eval.models.huggingface import HFLM
+from lm_eval.evaluator import evaluate
+from lm_eval.tasks import get_task_dict
+
+from torchao.quantization import quantize_, int8_weight_only, int4_weight_only, int8_dynamic_activation_int4_weight
+from torchao._models._eval import TransformerEvalWrapper
+
+from torchao.quantization.quant_primitives import (
+    MappingType,
+    ZeroPointDomain,
+)
+
+from torchao.quantization.quant_api import autoquant
+
+
+torch._inductor.config.force_fuse_int_mm_with_mul = True
+torch._inductor.config.fx_graph_cache = True
+
+
+def run_evaluation(repo_id, tasks, limit, device, precision, quantization, compile, batch_size, max_length, sensi_bit, non_sensi_bit, quant_sym, group_size):
+
+    tokenizer = AutoTokenizer.from_pretrained(repo_id)
+    model = AutoModelForCausalLM.from_pretrained(repo_id).to(device="cpu", dtype=precision)
+
+    if quantization == "int8dq":
+        quantize_(model.to(device=device), int8_dynamic_activation_int4_weight())
+
+    elif quantization == "int8wo":
+        quantize_(model.to(device=device), int8_weight_only())
+
+    elif quantization == "int4wo": 
+        quantize_(model.to(device=device), int4_weight_only(group_size=group_size))
+        
+    elif quantization == "autoquant":
+        model = autoquant(model.to(device=device))
+        
+    # naive implementation of uniform precision quantization all layers    
+    elif quantization in ["2","3","4","5","6","8"]:
+        if quant_sym == "asym":
+            quantize_(model.to(device=device), intN_weight_only_asym(n=int(quantization), group_size=group_size))
+        elif quant_sym == "sym":
+            quantize_(model.to(device=device), intN_weight_only_sym(n=int(quantization), group_size=group_size))
+        
+    elif quantization == "MP_llama3":
+        
+        # filter for sensitive layers
+        def filter_fn_sen(child: torch.nn.Module, cur_fqn:str) -> bool:
+            return isinstance(child, nn.Linear) and any(skiplayer in cur_fqn for skiplayer in ['.0.', '.1.', '.2.', '.30.', '.31.'])
+        
+        # filter for non-sensitive layers
+        def filter_fn_nonsen(child: torch.nn.Module, cur_fqn:str) -> bool:
+            return isinstance(child, nn.Linear) and not(any(skiplayer in cur_fqn for skiplayer in ['.0.', '.1.', '.2.', '.30.', '.31.']))
+        
+        if sensi_bit != 16:
+            # quantize the sensitive layers
+            if sensi_bit == 8:
+                quantize_(model.to(device=device), int8_weight_only(), filter_fn_sen)
+            elif  sensi_bit == 4: 
+                quantize_(model.to(device=device), int4_weight_only(group_size=group_size), filter_fn_sen)
```

Comment:
you could merge these logic into `intN_weight_only_asym` I think

Reply:
merged them into intN_weight_only now